packages/hash/frontend/src/components/layout/PageHeader/SearchBar/DesktopSearch.tsx

```diff
@@ -0,0 +1,147 @@
+import { alpha, Box, InputBase, styled } from "@mui/material";
+import { useRef } from "react";
+import { useKeys } from "rooks";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+
+import { SearchIcon, FontAwesomeSvgIcon } from "../../../icons";
+
+const SearchIconWrapper = styled("div")(({ theme }) => ({
+  padding: theme.spacing(0, 2),
+  height: "100%",
+  position: "absolute",
+  pointerEvents: "none",
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+}));
+
+const SlashIconWrapper = styled("div")(({ theme }) => ({
+  margin: theme.spacing(0, 1),
+  height: "100%",
+  position: "absolute",
+  pointerEvents: "none",
+  display: "flex",
+  alignItems: "center",
+  top: 0,
+  right: 0,
+  fontWeight: "bold",
+}));
+
+const CrossIconWrapper = styled("div")(({ theme }) => ({
+  margin: theme.spacing(0, 1),
+  marginRight: 0,
+  padding: theme.spacing(0, 1),
+  position: "absolute",
+  display: "flex",
+  height: "100%",
+  alignItems: "center",
+  cursor: "pointer",
+  top: 0,
+  right: 0,
+  fontWeight: "bold",
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+  color: "inherit",
+  "& .MuiInputBase-input": {
+    padding: theme.spacing(1, 0, 1, 0),
+    // vertical padding + font size from searchIcon
+    paddingLeft: `calc(1em + ${theme.spacing(3)})`,
+    paddingRight: `calc(1em + ${theme.spacing(3)})`,
+    transition: theme.transitions.create("width"),
+    [theme.breakpoints.up("md")]: {
+      width: "20ch",
+    },
+    border: `1px solid ${theme.palette.gray[30]}`,
+    borderRadius: "6px",
+    position: "relative",
+
+    ":focus": {
+      margin: "-1px",
+      border: `2px solid ${theme.palette.blue[70]}`,
+    },
+  },
+}));
+
+const Search = styled("div")(({ theme }) => ({
+  position: "relative",
+  borderRadius: theme.shape.borderRadius,
+  backgroundColor: alpha(theme.palette.common.white, 0.15),
+  "&:hover": {
+    backgroundColor: alpha(theme.palette.common.white, 0.25),
+  },
+  marginRight: theme.spacing(2),
+}));
```

Comment:
I think we should reserve usage of styled components for when we want to create re-usable styles. I don't think any of these are/will be used more than once, so can we make these inline styles instead?

If you're trying to reduce the size of the react tree, I'd recommend splitting up the hierarchy into separate components instead of using styled components here.

Reply:
Done here: https://github.com/hashintel/hash/commit/d0a35c2918755940192b1510e9ae086fd957d1c4